common/src/main/scala/org/neo4j/spark/writer/BaseDataWriter.scala

```diff
@@ -91,15 +93,25 @@ abstract class BaseDataWriter(jobId: String,
     Unit
   }
 
+  /**
+   * df: we check if the thrown exception is STOPPED_THREAD_EXCEPTION. This is the
+   * exception that is thrown when the streaming query is interrupted, we don't want to cause
+   * any error in this case. The transaction are rolled back automatically.
+   */
   private def logAndThrowException(e: Exception): Unit = {
-    if (e.isInstanceOf[ClientException]) {
-      log.error(s"Cannot commit the transaction because: ${e.getMessage}")
+    if(e.isInstanceOf[ServiceUnavailableException] && e.getMessage == STOPPED_THREAD_EXCEPTION_MESSAGE) {
```

Comment:
In this catch maybe makes sense check what kind of root cause has the exception, if is `org.apache.spark.TaskKilledException` we can log the message otherwise we throw the exception